api/v3/AwardImport.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * @file
+ * Award Import API file.
+ */
+
+use CRM_CiviAwards_Service_AwardImport as AwardImportService;
+
+/**
+ * AwardImport.create API specification.
+ *
+ * @param array $spec
+ *   Description of fields supported by this API call.
+ */
+function _civicrm_api3_award_import_create_spec(array &$spec) {
```

Comment:
I think what we need here for the fields is basically the Case Type fields and the Award Details Entity fields. We can leverage on Civi's API calls here

```php
  $caseTypeFields = civicrm_api3('CaseType', 'getfields', ['api_action' => 'get']);
  $awardDetailFields = civicrm_api3('AwardDetail', 'getfields', ['api_action' => 'get']);
  $spec = array_merge($caseTypeFields, $awardDetailFields)

```

ref: https://github.com/compucorp/uk.co.compucorp.civicase/blob/07b7746e2ad5af5fa6d5d3e74e3d74e9cc7f5c9f/api/v3/Case/Getdetails.php#L18

Reply:
I tried this, because sounds a good idea. I merged the values of the arrays, but I couldn't make it work finally.
The final array was similar to the one that I manually created, but with much more information on every row.
At the end, when I tried the process manually, it returned a 500 error, very difficult to debug (I guess because of the queue usage, was so difficult to see what was failing)
It was taking me too long, then I decided to move on with this. I removed the unused specs: https://github.com/compucorp/uk.co.compucorp.civiawards/pull/164/commits/fedf6f7677a992d1ef589724ea17e5b987d17849